### PR TITLE
Roger Guitart OptionPayout Refactoring - Definitive

### PR DESCRIPTION
**Pull request by hugohills-regnosys**
Replaces #2694 with updated expectations

### CODE REVIEW

**Review by minesh-s-patel — APPROVED** [review]

**Comment by lolabeis** [comment]
@rogerguitart Please include your release notes as a comment in this PR.

I have looked at the release notes you submitted in an earlier PR but they're missing the required _Backward-incompatible changes_ section. This section should contain a detailed set of instructions for implementors to remap any object persisted according to an earlier version into the new structure, ideally pointing to relevant JSON samples illustrating the changes. Please see: https://github.com/finos/common-domain-model/releases/tag/5.0.0 for a template.

**Comment by PayalKhanna** [comment]
@rogerguitart 
your release notes have been added to this PR please make amendments as requested by Leo directly [here](https://github.com/rosetta-models/common-domain-model/blob/Refactoring_OptionPayoutDefinitive-merge/RELEASE.md)

**Comment by rogerguitart** [comment]
@lolabeis I've updated the release notes to include the _Backward Incompatible Changes_ Section where @PayalKhanna has suggested. Following the template, I've added only 3 json examples of how samples would be modified (an american, a european, and a bermuda) to exemplify the changes, given that there are a large number of samples affected and the release notes would just be too long.

File: RELEASE.md [hunk 1]
```diff
@@ -1,3 +1,418 @@
...
+
+_What is being released?_
+
+- The `optionStyle` is removed from the model, along with the `americanExercise`, `europeanExercise`, and `bermudaExercise` fields.
+- A new `ExerciseTerms` structure is added to `OptionPayout`, containing all _distinct_ fields found previously under the three exercise styles.
+- A new `style` enumeration is added under `ExerciseTerms` to distinguish the style of the option. This enumeration is made **optional** to account the exercise terms of a `CancelableProvision`, `ExtendibleProvision`, or `OptionalEarlyTermination`, where the style is often derived from the structure itself.
```
Comment (lolabeis) [inline RELEASE.md:12]: "to account **for**"

File: RELEASE.md [hunk 2]
```diff
@@ -1,3 +1,418 @@
...
+_Enumerations_
+
+- Added new `OptionExerciseStyleEnum` enumeration.
+
+- _Backward Incompatible Changes_
+
+
+- _Motivation for Change_
+
+This change modifies the `OptionPayout` structure to simplify it and reduce the redundancy and over-complexity of the model.
```
Comment (lolabeis) [inline RELEASE.md:37]: Remove this "motivation for change" section given the "background" section in preamble.

File: RELEASE.md [hunk 3]
```diff
@@ -1,3 +1,418 @@
...
+        "exerciseTerms": {
+            "optionStyle": {
+                "europeanExercise": {
+                    "expirationDate": [
+                        {
+                            "adjustableDate": {
+                                "unadjustedDate": "2004-12-19",
+                                "dateAdjustments": {
+                                    "businessDayConvention": "NONE",
+                                    "meta": {"globalKey": "24a738"}
```
Comment (lolabeis) [inline RELEASE.md:66]: Remove all meta / key from the JSON samples, they pollute the view and aren't necessary

File: RELEASE.md [hunk 4]
```diff
@@ -1,3 +1,418 @@
...
+                    "exerciseNotice": {
+                        "exerciseNoticeGiver": "Seller",
+                        "businessCenter": {"value": "GBLO"}
+                    }
+                },
+                "followUpConfirmation": true
+            },
+            "meta": {
+                "globalKey": "5c12e2ce",
+                "externalKey": "bermudaExercise0"
```
Comment (lolabeis) [inline RELEASE.md:403]: Just keep this one "meta" as it's the only one needed for referential integrity

**Comment by lolabeis** [comment]
@rogerguitart thanks for completing the backward incompatible section.
Please adjust the RN as suggested and I will approve, this PR is now good to go.

**Comment by rogerguitart** [comment]
Release notes have been amended - please check @lolabeis

**Pull request merged**